goal_src/levels/misty/misty-conveyor.gc

```diff
@@ -5,3 +5,1053 @@
+    (set!
+     (-> self state-time)
+     (the-as seconds (-> *display* base-frame-counter))
+     )
+    (suspend)
+    )
+   (none)
+   )
+  :post
+  (the-as (function none :behavior keg) keg-post)
+  )
+
+(defstate keg-paddle-to-path (keg)
+  :event
+  keg-event-handler
+  :code
+  (behavior ()
+   (let ((gp-0 (new-stack-vector0)))
+    (set! (-> gp-0 quad) (-> self root-override trans quad))
+    (let
+     ((s5-0
+       (eval-path-curve-div!
+        (-> (the-as process-drawable (-> self parent 0)) path)
+        (new-stack-vector0)
+        0.0
+        'interp
+        )
+       )
+      (s4-0
+       (quaternion-copy! (new-stack-quaternion0) (-> self root-override quat))
+       )
+      (s3-0 (new-stack-quaternion0))
+      (a1-3
+       (TODO-RENAME-12
+        (-> (the-as process-drawable (-> self parent 0)) path)
+        (new-stack-vector0)
+        0.0
+        )
+       )
+      (f30-0 45.0)
+      )
+     0.0
+     0.0
+     (forward-up-nopitch->quaternion
+      s3-0
+      a1-3
+      (new 'static 'vector :y 1.0 :w 1.0)
+      )
+     (while #t
+      (if
+       (>=
+        (the
+         float
+         (- (-> *display* base-frame-counter) (the-as int (-> self state-time)))
+         )
+        f30-0
+        )
+       (go keg-on-path)
+       )
+      (let
+       ((f28-0
+         (/
+          (the
+           float
+           (-
+            (-> *display* base-frame-counter)
+            (the-as int (-> self state-time))
+            )
+           )
+          f30-0
+          )
+         )
+        )
+       (vector-lerp! (-> self root-override trans) gp-0 s5-0 f28-0)
+       (set! (-> self path-position quad) (-> self root-override trans quad))
+       (quaternion-slerp! (-> self root-override quat) s4-0 s3-0 f28-0)
+       )
+      (let ((a0-11 (-> self skel root-channel 0)))
+       (set! (-> a0-11 param 0) 1.0)
+       (joint-control-channel-group-eval!
+        a0-11
+        (the-as art-joint-anim #f)
+        num-func-loop!
+        )
+       )
+      (suspend)
+      )
+     )
+    )
+   (none)
+   )
+  :post
+  (the-as (function none :behavior keg) keg-post)
+  )
+
+(defstate keg-on-path (keg)
+  :event
+  keg-event-handler
+  :code
+  (behavior ()
+   (local-vars
+    (sv-48 float)
+    (sv-64 float)
+    (sv-80 float)
+    (sv-96 float)
+    (sv-112 float)
+    )
+   (let ((gp-0 (new-stack-vector0))
+         (s5-0 (new 'stack 'vector3s))
+         )
+    0.0
+    (let ((f30-0 0.0))
+     0.0
+     (let
+      ((f28-0
+        (the
+         float
+         (+
+          (->
+           (the-as process-drawable (-> self parent 0))
+           path
+           curve
+           num-cverts
+           )
+          -1
+          )
+         )
+        )
+       )
+      0.0
+      (let
+       ((f26-1
+         (-
+          1.0
+          ((method-of-type sync-info get-current-phase)
+           (the-as
+            sync-info
+            (&-> (the-as process-drawable (-> self parent 0)) stack 68)
```

Comment:
I think `parent` isn't a `process-drawable` here but instead a `keg-conveyor-paddle`.  Hopefully that would get you the `sync-info`